SSA/Experimental/Bits/Fast/BitStream.lean

```diff
@@ -418,8 +418,37 @@ theorem not_congr (e1 : a ≈ʷ b) : (~~~a) ≈ʷ ~~~b := by
   intros g h
   simp only [not_eq, e1 g h]
 
+theorem ofBitVec_not_eq_to : ofBitVec (~~~ x) ≈ʷ ~~~ ofBitVec x := by
+  intros _ a
+  simp [ofBitVec, a]
+
+theorem negAux_eq_not_addAux : a.negAux = Prod.swap ∘ (~~~a).addAux 1 := by
+  funext i
+  induction' i with _ ih
```

Comment:
Minor nit: it wouldn't hurt to have some explanation of the `Prod.swap` in a docstring for this theorem

Reply:
It is very strange: in negAux, the carry bit is the right element of the pair, but in addAux, the carry bit is the left element of the pair. I don't know why this is.

Reply:
Perhaps @tobiasgrosser could comment as to why this is, as it appears that he wrote that code.

Reply:
That's what I figured, I meant just write exactly that, just write "We need `Prod.swap` here because in negAux, the carry bit is the right element of the pair, but in addAux it's the left element" as the docstring